bindings.go

```diff
@@ -0,0 +1,181 @@
+		if err != nil {
+			return reflect.TypeOf(nil), err
+		}
+		return reflect.MapOf(kType, vType), nil
+	case TypeSet:
+		kType, err := nativeTypeFromExtended(column.TypeObj.Key.Type)
+		if err != nil {
+			return reflect.TypeOf(nil), err
+		}
+		return reflect.SliceOf(kType), nil
+	default:
+		return reflect.TypeOf(nil), fmt.Errorf("Failed to determine type of column %s", column)
+	}
+}
+
+// OvsToNative transforms an ovs type to native one based on the column type information
+func OvsToNative(column *ColumnSchema, ovsElem interface{}) (interface{}, error) {
+	naType, err := NativeType(column)
+	if err != nil {
+		return nil, err
+	}
+	switch column.Type {
+	case TypeInteger, TypeReal, TypeString, TypeBoolean, TypeEnum:
+		// Atomic types should have the same underlying type
+		return ovsElem, nil
+	case TypeUUID:
+		uuid, ok := ovsElem.(UUID)
+		if !ok {
+			return nil, fmt.Errorf("Element in column should be convertible to UUID. Instead got %s", reflect.TypeOf(ovsElem))
+		}
+		return uuid.GoUUID, nil
+	case TypeSet:
+		ovsSet, ok := ovsElem.(OvsSet)
+		if !ok {
+			return nil, fmt.Errorf("Element in column should be convertible to Set. Instead got %s", reflect.TypeOf(ovsElem))
+		}
+		// The inner slice is []interface{}
+		// We need to convert it to the real type os slice
+		nativeSet := reflect.MakeSlice(naType, 0, len(ovsSet.GoSet))
+		for _, v := range ovsSet.GoSet {
+			vv, err := NativeValueOf(v, column.TypeObj.Key.Type)
+			if err != nil {
+				return nil, err
+			}
+			nativeSet = reflect.Append(nativeSet, vv)
+		}
+		return nativeSet.Interface(), nil
+	case TypeMap:
+		ovsMap, ok := ovsElem.(OvsMap)
+		if !ok {
+			return nil, fmt.Errorf("Element in column should be convertible to Map. Instead got %s", reflect.TypeOf(ovsElem))
+		}
+		// The inner slice is map[interface]interface{}
+		// We need to convert it to the real type os slice
+		nativeMap := reflect.MakeMapWithSize(naType, len(ovsMap.GoMap))
+		for k, v := range ovsMap.GoMap {
+			kk, err := NativeValueOf(k, column.TypeObj.Key.Type)
+			if err != nil {
+				return nil, err
+			}
+			vv, err := NativeValueOf(v, column.TypeObj.Value.Type)
+			if err != nil {
+				return nil, err
+			}
+			nativeMap.SetMapIndex(kk, vv)
+		}
+		return nativeMap.Interface(), nil
+	default:
+		return nil, fmt.Errorf("Unknown type %s", column.Type)
+	}
+}
+
+// NativeToOvs transforms an native type to a ovs type based on the column type information
+func NativeToOvs(column *ColumnSchema, rawElem interface{}) (interface{}, error) {
+	naType, err := NativeType(column)
+	if err != nil {
+		return nil, err
+	}
+	if t := reflect.TypeOf(rawElem); t != naType {
+		return nil, fmt.Errorf("Bad Type in column expected %s, got %s (%v)",
+			naType.String(), t.String(), rawElem)
+	}
+
+	switch column.Type {
+	case TypeInteger, TypeReal, TypeString, TypeBoolean, TypeEnum:
+		return rawElem, nil
+	case TypeUUID:
+		return UUID{GoUUID: rawElem.(string)}, nil
+	case TypeSet:
```

Comment:
For a column with type "Set", rawElem can be a single atomic element ...

I saw this after the above comment: https://github.com/eBay/libovsdb/pull/30/commits/c8f8ff3946284b9474419e89352ad2d8976f7410

It is better to squash them together as it is a fix to this commit.

Reply:
Sure, will do